projects/storefrontlib/src/cms-components/user/login-form/login-form.component.ts

```diff
@@ -42,26 +48,24 @@ export class LoginFormComponent implements OnInit, OnDestroy {
         'forced'
       ];
     }
+  }
 
-    const prefilledEmail = this.winRef?.nativeWindow?.history?.state?.[
-      'newUid'
-    ];
-
-    if (prefilledEmail?.length) {
-      this.prefillForm('userId', prefilledEmail);
+  submitForm(): void {
+    if (this.loginForm.valid) {
+      this.login();
+    } else {
+      this.loginForm.markAllAsTouched();
     }
   }
 
-  login(): void {
-    if (this.form.valid) {
-      this.submitLogin();
-    } else {
-      this.markFormAsTouched();
+  ngOnDestroy(): void {
+    if (this.sub) {
+      this.sub.unsubscribe();
     }
   }
 
-  private submitLogin(): void {
-    const { userId, password } = this.form.controls;
+  private login(): void {
```

Comment:
I would make it protected to improve extensibility and change name to not confuse about the previous implementation which did a different thing.

Reply:
Done